File: core/src/main/java/top/iseason/bukkittemplate/utils/JavaVersion.java
package top.iseason.bukkittemplate.utils;

public class JavaVersion {
    private static int version;

    static {
        String versionString = System.getProperty("java.version");
        int indexOf = versionString.indexOf('.');
        try {
            if (indexOf > 0) {
                String substring = versionString.substring(0, indexOf);
                if (substring.equals("1")) {
                    int indexOf1 = versionString.indexOf('.', indexOf + 1);
                    version = Integer.parseInt(versionString.substring(indexOf + 1, indexOf1));
                } else {
                    version = Integer.parseInt(substring);
                }
            } else {
                version = Integer.parseInt(versionString);
            }
        } catch (Exception e) {
            version = 8;
        }
    }

    public static boolean isGreaterOrEqual(int version) {
        return JavaVersion.version >= version;
    }
}

File: core/src/main/kotlin/top/iseason/bukkittemplate/utils/JavaVersion.kt
package top.iseason.bukkittemplate.utils

object JavaVersion {
    val version: Int

    init {
        val versionString = System.getProperty("java.version")
        val indexOf = versionString.indexOf('.')
        version = try {
            if (indexOf > 0) {
                val substring = versionString.substring(0, indexOf)
                if (substring == "1") {
                    val indexOf1 = versionString.indexOf('.', indexOf + 1)
                    versionString.substring(indexOf + 1, indexOf1).toInt()
                } else {
                    substring.toInt()
                }
            } else {
                versionString.toInt()
            }
        } catch (_: Exception) {
            8
        }
    }

    fun isGreaterOrEqual(version: Int): Boolean = this.version >= version

}